eval/eval.go

```diff
@@ -672,6 +673,124 @@ func (s *State) evalIfExpression(ie *ast.IfExpression) object.Object {
 	}
 }
 
+func (s *State) evalForInteger(fe *ast.ForExpression, start *object.Integer, end object.Integer, name string) object.Object {
+	var lastEval object.Object
+	lastEval = object.NULL
+	startValue := 0
+	if start != nil {
+		startValue = int(start.Value)
+	}
+	endValue := int(end.Value)
+	num := endValue - startValue
+	if num < 0 {
+		return s.Errorf("for loop with negative count [%d,%d[", startValue, endValue)
+	}
+	for i := startValue; i < endValue; i++ {
+		if name != "" {
+			s.env.Set(name, object.Integer{Value: int64(i)})
+		}
+		lastEval = s.evalInternal(fe.Body)
+		if rt := lastEval.Type(); rt == object.RETURN || rt == object.ERROR {
+			return lastEval
+		}
+	}
+	return lastEval
+}
+
+func (s *State) evalForSpecialForms(fe *ast.ForExpression) (object.Object, bool) {
+	ie, ok := fe.Condition.(*ast.InfixExpression)
+	if !ok {
+		return object.NULL, false
+	}
+	if ie.Token.Type() != token.ASSIGN {
+		return object.NULL, false
+	}
+	if ie.Left.Value().Type() != token.IDENT {
+		return s.Errorf("for var = ... not a var %s", ie.Left.Value().DebugString()), true
```

Comment:
Do you have a kind of language words you forbid as syntax variable.

Something like for, and, or, 

It would avoid strange syntax such as

for for = [0:5]

Reply:
well `for for`does fail because `for` is a keyword

reserved are
```
$ info.keywords
["else","false","for","func","if","return","true"]
$ info.builtins
["error","first","len","log","macro","print","println","quote","rest","unquote"]
```